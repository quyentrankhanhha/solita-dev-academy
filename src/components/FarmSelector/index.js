import { Box, FormControl, InputLabel, NativeSelect } from '@mui/material'
import React from 'react'
import { period_type, sensor_type } from '../../constant'

const FarmSelector = ({
  farms,
  farmValue,
  periodValue,
  sensorValue,
  handleOnFarmChange,
  handleOnPeriodChange,
  handleOnSensorChange,
}) => {
  return (
    <Box>
      <FormControl>
        <InputLabel htmlFor='' shrink>
          Farm
        </InputLabel>
        <NativeSelect
          value={farmValue}
          onChange={handleOnFarmChange}
          inputProps={{
            name: 'farm',
            id: 'farm-selector',
          }}
        >
          {farms?.map((farm, index) => {
            return (
              <option value={farm.farm_id} key={`farm-${index}`}>
                {farm?.name}
              </option>
            )
          })}
        </NativeSelect>
      </FormControl>
      <FormControl>
        <InputLabel htmlFor='' shrink>
          Period
        </InputLabel>
        <NativeSelect
          value={periodValue}
          onChange={handleOnPeriodChange}
          inputProps={{
            name: 'period',
            id: 'period-selector',
          }}
        >
          {period_type?.map((type, index) => {
            return (
              <option value={type.type} key={`farm-${index}`}>
                {type?.name}
              </option>
            )
          })}
        </NativeSelect>
      </FormControl>
      {periodValue !== 'all' && (
        <FormControl>
          <InputLabel htmlFor='' shrink>
            Sensor Type
          </InputLabel>
          <NativeSelect
            value={sensorValue}
            onChange={handleOnSensorChange}
            inputProps={{
              name: 'sensor',
              id: 'sensor-selector',
            }}
          >
            {sensor_type?.map((type, index) => {
              return (
                <option value={type} key={`farm-${index}`}>
                  {type}
                </option>
              )
            })}
          </NativeSelect>
        </FormControl>
      )}
    </Box>
  )
}

export default FarmSelector
