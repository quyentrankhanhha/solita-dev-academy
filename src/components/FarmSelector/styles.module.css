.selector_wrapper {
  padding: 15px 0;
}

.selector {
  padding: 0 10px !important;
}
