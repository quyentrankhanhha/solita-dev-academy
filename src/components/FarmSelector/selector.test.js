import { fireEvent, render, screen } from '@testing-library/react'
import FarmSelector from './index'

const farms = [
  {
    farm_id: '1',
    location: 'Metsola',
    name: 'Friman Metsola Collective',
  },
]
test('render component', () => {
  const { container } = render(
    <FarmSelector
      farms={farms}
      handleOnFarmChange={() => {}}
      handleOnPeriodChange={() => {}}
      handleOnSensorChange={() => {}}
      farmValue={() => {}}
      periodValue={() => {}}
      sensorValue={() => {}}
      handleOnChangeType={() => {}}
      handleOnFarmChoice={() => {}}
    />
  )
  expect(container).toMatchSnapshot()
})

jest.mock('react-select', () => ({ options, value, onChange }) => {
  function handleChange(event) {
    const option = options.find(
      (option) => option.value === event.currentTarget.value
    )
    onChange(option)
  }

  return (
    <select data-testid='select' value={value} onChange={handleChange}>
      {options.map(({ label, value }) => (
        <option key={value} value={value}>
          {label}
        </option>
      ))}
    </select>
  )
})

test('should be have sensor selector like ph sensor when choosing a month period', () => {
  render(<FarmSelector />)
  expect(screen.getByText('A Month')).toBeInTheDocument()
  fireEvent.change(screen.getByTestId('select-period'), {
    target: { value: 'month' },
  })
  // highlight-end
  expect(screen.getByText('Temperature')).toBeInTheDocument()
})
