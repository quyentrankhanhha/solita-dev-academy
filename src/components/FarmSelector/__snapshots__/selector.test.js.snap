// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FarmSelector render component 1`] = `
<div>
  <div
    class="selector_wrapper MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=""
      >
        Farm
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          id="farm-selector"
          name="farm"
        >
          <option
            value="1"
          >
            Friman Metsola Collective
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root selector css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="period"
      >
        Period
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          data-testid="select-period"
          id="period"
          name="period"
        >
          <option
            data-testid="period-selection"
            value="all"
          >
            All
          </option>
          <option
            data-testid="period-selection"
            value="month"
          >
            A Month
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="sensor"
      >
        Sensor Type
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          id="sensor"
          name="sensor"
        >
          <option
            value="none"
          >
            Choose sensor
          </option>
          <option
            value="temperature"
          >
            Temperature
          </option>
          <option
            value="ph"
          >
            pH
          </option>
          <option
            value="rainfall"
          >
            Rainfall
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="btn_container"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        add data
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`render component 1`] = `
<div>
  <div
    class="selector_wrapper MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=""
      >
        Farm
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          id="farm-selector"
          name="farm"
        >
          <option
            value="1"
          >
            Friman Metsola Collective
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root selector css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="period"
      >
        Period
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          data-testid="select-period"
          id="period"
          name="period"
        >
          <option
            data-testid="period-selection"
            value="all"
          >
            All
          </option>
          <option
            data-testid="period-selection"
            value="month"
          >
            A Month
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="sensor"
      >
        Sensor Type
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <select
          aria-invalid="false"
          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInput-input MuiInputBase-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
          id="sensor"
          name="sensor"
        >
          <option
            value="none"
          >
            Choose sensor
          </option>
          <option
            value="temperature"
          >
            Temperature
          </option>
          <option
            value="ph"
          >
            pH
          </option>
          <option
            value="rainfall"
          >
            Rainfall
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="btn_container"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        add data
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
